### PR TITLE
Allow bracket IPv6 address format inside IPAdress Normalizer

### LINKED ISSUE #13943: IP [::1] makes the instance unusable
### Steps to reproduce
1. Set up a new instance with the php built-in server (php -S localhost:8080)
2. Complete the setup page

### Expected behaviour
Have a working instance

### Actual behaviour
Fatal error. My address is shown as `[::1]`.

```json


{
  "Exception": "TypeError",
  "Message": "inet_ntop() expects parameter 1 to be string, bool given",
  "Code": 0,
  "Trace": [
    {
      "file": "\/tmp\/nextcloud\/lib\/private\/Security\/Normalizer\/IpAddress.php",
      "line": 79,
      "function": "inet_ntop",
      "args": [
        false
      ]
    },
    {
      "file": "\/tmp\/nextcloud\/lib\/private\/Security\/Normalizer\/IpAddress.php",
      "line": 96,
      "function": "getIPv6Subnet",
      "class": "OC\\Security\\Normalizer\\IpAddress",
      "type": "->",
      "args": [
        "[::1]",
        128
      ]
    },
    {
      "file": "\/tmp\/nextcloud\/lib\/private\/Security\/Bruteforce\/Throttler.php",
      "line": 219,
      "function": "getSubnet",
      "class": "OC\\Security\\Normalizer\\IpAddress",
      "type": "->",
      "args": [
        
      ]
    },
    {
      "file": "\/tmp\/nextcloud\/lib\/private\/Security\/Bruteforce\/Capabilities.php",
      "line": 50,
      "function": "getDelay",
      "class": "OC\\Security\\Bruteforce\\Throttler",
      "type": "->",
      "args": [
        "[::1]"
      ]
    },
    {
      "file": "\/tmp\/nextcloud\/lib\/private\/CapabilitiesManager.php",
      "line": 68,
      "function": "getCapabilities",
      "class": "OC\\Security\\Bruteforce\\Capabilities",
      "type": "->",
      "args": [
        
      ]
    },
    {
      "file": "\/tmp\/nextcloud\/lib\/private\/Template\/JSConfigHelper.php",
      "line": 166,
      "function": "getCapabilities",
      "class": "OC\\CapabilitiesManager",
      "type": "->",
      "args": [
        
      ]
    },
    {
      "file": "\/tmp\/nextcloud\/lib\/private\/TemplateLayout.php",
      "line": 170,
      "function": "getConfig",
      "class": "OC\\Template\\JSConfigHelper",
      "type": "->",
      "args": [
        
      ]
    },
    {
      "file": "\/tmp\/nextcloud\/lib\/private\/legacy\/template.php",
      "line": 188,
      "function": "__construct",
      "class": "OC\\TemplateLayout",
      "type": "->",
      "args": [
        "user",
        "files"
      ]
    },
    {
      "file": "\/tmp\/nextcloud\/lib\/public\/AppFramework\/Http\/TemplateResponse.php",
      "line": 157,
      "function": "fetchPage",
      "class": "OC_Template",
      "type": "->",
      "args": [
        {
          "usedSpacePercent": 0,
          "owner": "admin",
          "ownerDisplayName": "admin",
          "isPublic": false,
          "allowShareWithLink": "yes",
          "defaultFileSorting": "name",
          "defaultFileSortingDirection": "asc",
          "showgridview": false,
          "isIE": false,
          "showHiddenFiles": 0,
          "fileNotFound": 0,
          "appNavigation": {
            "__class__": "OCP\\Template"
          },
          "appContents": {
            "files": {
              "id": "files",
              "content": "<div id=\"controls\">\n\t\t<div class=\"actions creatable hidden\">\n\t\t\t<div id=\"uploadprogresswrapper\">\n\t\t\t\t<div id=\"uploadprogressbar\">\n\t\t\t\t\t<em class=\"label outer\" style=\"display:none\"><span class=\"desktop\">Uploading \u2026<\/span><span class=\"mobile\">\u2026<\/span><\/em>\n\t\t\t\t<\/div>\n\t\t\t\t<button class=\"stop icon-close\" style=\"display:none\">\n\t\t\t\t\t<span class=\"hidden-visually\">Cancel upload<\/span>\n\t\t\t\t<\/button>\n\t\t\t<\/div>\n\t\t<\/div>\n\t\t<div id=\"file_action_panel\"><\/div>\n\t\t<div class=\"notCreatable notPublic hidden\">\n\t\t\tYou don\u2019t have permission to upload or create files here\t\t<\/div>\n\t\t<input type=\"hidden\" name=\"permissions\" value=\"\" id=\"permissions\">\n\t<input type=\"hidden\" id=\"free_space\" value=\"\">\n\t\t<input type=\"hidden\" class=\"max_human_file_size\"\n\t\t   value=\"(max )\">\n<\/div>\n\n<div id=\"emptycontent\" class=\"hidden\">\n\t<div class=\"icon-folder\"><\/div>\n\t<h2>No files in here<\/h2>\n\t<p class=\"uploadmessage hidden\">Upload some content or sync with your devices!<\/p>\n<\/div>\n\n<div class=\"nofilterresults emptycontent hidden\">\n\t<div class=\"icon-search\"><\/div>\n\t<h2>No entries found in this folder<\/h2>\n\t<p><\/p>\n<\/div>\n<table id=\"filestable\" class=\"list-container \" data-allow-public-upload=\"yes\" data-preview-x=\"250\" data-preview-y=\"250\">\n\t<thead>\n\t\t<tr>\n\t\t\t<th id=\"headerSelection\" class=\"hidden column-selection\">\n\t\t\t\t<input type=\"checkbox\" id=\"select_all_files\" class=\"select-all checkbox\"\/>\n\t\t\t\t<label for=\"select_all_files\">\n\t\t\t\t\t<span class=\"hidden-visually\">Select all<\/span>\n\t\t\t\t<\/label>\n\t\t\t<\/th>\n\t\t\t<th id='headerName' class=\"hidden column-name\">\n\t\t\t\t<div id=\"headerName-container\">\n\t\t\t\t\t<a class=\"name sort columntitle\" data-sort=\"name\">\n                        <span>Name<\/span>\n                        <span class=\"sort-indicator\"><\/span>\n\n                    <\/a>\n                    <span id=\"selectedActionsList\" class=\"selectedActions\">\n                        <a href=\"\" class=\"actions-selected\">\n                            <span class=\"icon icon-more\"><\/span>\n                            <span>Actions<\/span>\n                        <\/a>\n\t\t\t\t\t<\/span>\n\t\t\t\t<\/div>\n\t\t\t<\/th>\n\t\t\t<th id=\"headerSize\" class=\"hidden column-size\">\n\t\t\t\t<a class=\"size sort columntitle\" data-sort=\"size\"><span>Size<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t<\/th>\n\t\t\t<th id=\"headerDate\" class=\"hidden column-mtime\">\n\t\t\t\t<a id=\"modified\" class=\"columntitle\" data-sort=\"mtime\"><span>Modified<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t<\/th>\n\t\t<\/tr>\n\t<\/thead>\n\t<tbody id=\"fileList\">\n\t<\/tbody>\n\t<tfoot>\n\t<\/tfoot>\n<\/table>\n<input type=\"hidden\" name=\"dir\" id=\"dir\" value=\"\" \/>\n<div class=\"hiddenuploadfield\">\n\t<input type=\"file\" id=\"file_upload_start\" class=\"hiddenuploadfield\" name=\"files[]\" \/>\n<\/div>\n<div id=\"editor\"><\/div><!-- FIXME Do not use this div in your app! It is deprecated and will be removed in the future! -->\n<div id=\"uploadsize-message\" title=\"Upload too large\">\n\t<p>\n\tThe files you are trying to upload exceed the maximum size for file uploads on this server.\t<\/p>\n<\/div>\n"
            },
            "recent": {
              "id": "recent",
              "content": "<div id='notification'><\/div>\n\n<div id=\"emptycontent\" class=\"hidden\"><\/div>\n\n<input type=\"hidden\" name=\"dir\" value=\"\" id=\"dir\">\n\n<div class=\"nofilterresults hidden\">\n\t<div class=\"icon-search\"><\/div>\n\t<h2>No entries found in this folder<\/h2>\n\t<p><\/p>\n<\/div>\n\n<table id=\"filestable\" class=\"list-container \">\n\t<thead>\n\t<tr>\n\t\t<th id='headerName' class=\"hidden column-name\">\n\t\t\t<div id=\"headerName-container\">\n\t\t\t\t<a class=\"name sort columntitle\"\n\t\t\t\t   data-sort=\"name\"><span>Name<\/span><\/a>\n\t\t\t<\/div>\n\t\t<\/th>\n\t\t<th id=\"headerSize\" class=\"hidden column-size\">\n\t\t\t<a class=\"size sort columntitle\"\n\t\t\t   data-sort=\"size\"><span>Size<\/span><\/a>\n\t\t<\/th>\n\t\t<th id=\"headerDate\" class=\"hidden column-mtime\">\n\t\t\t<a id=\"modified\" class=\"columntitle\"\n\t\t\t   data-sort=\"mtime\"><span>Modified<\/span><span\n\t\t\t\t\tclass=\"sort-indicator\"><\/span><\/a>\n\t\t\t<span class=\"selectedActions\">\n\t\t\t\t<a href=\"\" class=\"delete-selected\">\n\t\t\t\t\t<span class=\"icon icon-delete\"><\/span>\n\t\t\t\t\t<span>Delete<\/span>\n\t\t\t\t<\/a>\n\t\t\t<\/span>\n\t\t<\/th>\n\t<\/tr>\n\t<\/thead>\n\t<tbody id=\"fileList\">\n\t<\/tbody>\n\t<tfoot>\n\t<\/tfoot>\n<\/table>\n"
            },
            "favorites": {
              "id": "favorites",
              "content": "<div id='notification'><\/div>\n\n<div id=\"emptycontent\" class=\"hidden\">\n\t<div class=\"icon-starred\"><\/div>\n\t<h2>No favorites yet<\/h2>\n\t<p>Files and folders you mark as favorite will show up here<\/p>\n<\/div>\n\n<input type=\"hidden\" name=\"dir\" value=\"\" id=\"dir\">\n\n<div class=\"nofilterresults hidden\">\n\t<div class=\"icon-search\"><\/div>\n\t<h2>No entries found in this folder<\/h2>\n\t<p><\/p>\n<\/div>\n<table id=\"filestable\" class=\"list-container \">\n\t<thead>\n\t\t<tr>\n\t\t\t<th id='headerName' class=\"hidden column-name\">\n\t\t\t\t<div id=\"headerName-container\">\n\t\t\t\t\t<a class=\"name sort columntitle\" data-sort=\"name\"><span>Name<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t\t<\/div>\n\t\t\t<\/th>\n\t\t\t<th id=\"headerSize\" class=\"hidden column-size\">\n\t\t\t\t<a class=\"size sort columntitle\" data-sort=\"size\"><span>Size<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t<\/th>\n\t\t\t<th id=\"headerDate\" class=\"hidden column-mtime\">\n\t\t\t\t<a id=\"modified\" class=\"columntitle\" data-sort=\"mtime\"><span>Modified<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t\t<span class=\"selectedActions\">\n\t\t\t\t    <a href=\"\" class=\"delete-selected\">\n\t\t\t\t\t<img class=\"svg\" alt=\"\"\n\t\t\t\t\t     src=\"\/core\/img\/actions\/delete.svg\" \/>\n\t\t\t\t\tDelete\t\t\t\t    <\/a>\n\t\t\t\t<\/span>\n\t\t\t<\/th>\n\t\t<\/tr>\n\t<\/thead>\n\t<tbody id=\"fileList\">\n\t<\/tbody>\n\t<tfoot>\n\t<\/tfoot>\n<\/table>\n"
            },
            "sharingout": {
              "id": "sharingout",
              "content": "<div id='notification'><\/div>\n\n<div id=\"emptycontent\" class=\"hidden\"><\/div>\n\n<input type=\"hidden\" name=\"dir\" value=\"\" id=\"dir\">\n\n<div class=\"nofilterresults hidden\">\n\t<div class=\"icon-search\"><\/div>\n\t<h2>No entries found in this folder<\/h2>\n<\/div>\n\n<table id=\"filestable\" class=\"list-container \">\n\t<thead>\n\t\t<tr>\n\t\t\t<th id='headerName' class=\"hidden column-name\">\n\t\t\t\t<div id=\"headerName-container\">\n\t\t\t\t\t<a class=\"name sort columntitle\" data-sort=\"name\"><span>Name<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t\t<\/div>\n\t\t\t<\/th>\n\t\t\t<th id=\"headerDate\" class=\"hidden column-mtime\">\n\t\t\t\t<a id=\"modified\" class=\"columntitle\" data-sort=\"mtime\"><span>Share time<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t<\/th>\n\t\t\t<th class=\"hidden column-expiration\">\n\t\t\t\t<a class=\"columntitle\"><span>Expiration date<\/span><\/a>\n\t\t\t<\/th>\n\t\t<\/tr>\n\t<\/thead>\n\t<tbody id=\"fileList\">\n\t<\/tbody>\n\t<tfoot>\n\t<\/tfoot>\n<\/table>\n"
            },
            "sharingin": {
              "id": "sharingin",
              "content": "<div id='notification'><\/div>\n\n<div id=\"emptycontent\" class=\"hidden\"><\/div>\n\n<input type=\"hidden\" name=\"dir\" value=\"\" id=\"dir\">\n\n<div class=\"nofilterresults hidden\">\n\t<div class=\"icon-search\"><\/div>\n\t<h2>No entries found in this folder<\/h2>\n<\/div>\n\n<table id=\"filestable\" class=\"list-container \">\n\t<thead>\n\t\t<tr>\n\t\t\t<th id='headerName' class=\"hidden column-name\">\n\t\t\t\t<div id=\"headerName-container\">\n\t\t\t\t\t<a class=\"name sort columntitle\" data-sort=\"name\"><span>Name<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t\t<\/div>\n\t\t\t<\/th>\n\t\t\t<th id=\"headerDate\" class=\"hidden column-mtime\">\n\t\t\t\t<a id=\"modified\" class=\"columntitle\" data-sort=\"mtime\"><span>Share time<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t<\/th>\n\t\t\t<th class=\"hidden column-expiration\">\n\t\t\t\t<a class=\"columntitle\"><span>Expiration date<\/span><\/a>\n\t\t\t<\/th>\n\t\t<\/tr>\n\t<\/thead>\n\t<tbody id=\"fileList\">\n\t<\/tbody>\n\t<tfoot>\n\t<\/tfoot>\n<\/table>\n"
            },
            "sharinglinks": {
              "id": "sharinglinks",
              "content": "<div id='notification'><\/div>\n\n<div id=\"emptycontent\" class=\"hidden\"><\/div>\n\n<input type=\"hidden\" name=\"dir\" value=\"\" id=\"dir\">\n\n<div class=\"nofilterresults hidden\">\n\t<div class=\"icon-search\"><\/div>\n\t<h2>No entries found in this folder<\/h2>\n<\/div>\n\n<table id=\"filestable\" class=\"list-container \">\n\t<thead>\n\t\t<tr>\n\t\t\t<th id='headerName' class=\"hidden column-name\">\n\t\t\t\t<div id=\"headerName-container\">\n\t\t\t\t\t<a class=\"name sort columntitle\" data-sort=\"name\"><span>Name<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t\t<\/div>\n\t\t\t<\/th>\n\t\t\t<th id=\"headerDate\" class=\"hidden column-mtime\">\n\t\t\t\t<a id=\"modified\" class=\"columntitle\" data-sort=\"mtime\"><span>Share time<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t<\/th>\n\t\t\t<th class=\"hidden column-expiration\">\n\t\t\t\t<a class=\"columntitle\"><span>Expiration date<\/span><\/a>\n\t\t\t<\/th>\n\t\t<\/tr>\n\t<\/thead>\n\t<tbody id=\"fileList\">\n\t<\/tbody>\n\t<tfoot>\n\t<\/tfoot>\n<\/table>\n"
            },
            "deletedshares": {
              "id": "deletedshares",
              "content": "<div id='notification'><\/div>\n\n<div id=\"emptycontent\" class=\"hidden\"><\/div>\n\n<input type=\"hidden\" name=\"dir\" value=\"\" id=\"dir\">\n\n<div class=\"nofilterresults hidden\">\n\t<div class=\"icon-search\"><\/div>\n\t<h2>No entries found in this folder<\/h2>\n<\/div>\n\n<table id=\"filestable\" class=\"list-container \">\n\t<thead>\n\t\t<tr>\n\t\t\t<th id='headerName' class=\"hidden column-name\">\n\t\t\t\t<div id=\"headerName-container\">\n\t\t\t\t\t<a class=\"name sort columntitle\" data-sort=\"name\"><span>Name<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t\t<\/div>\n\t\t\t<\/th>\n\t\t\t<th id=\"headerDate\" class=\"hidden column-mtime\">\n\t\t\t\t<a id=\"modified\" class=\"columntitle\" data-sort=\"mtime\"><span>Share time<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t<\/th>\n\t\t\t<th class=\"hidden column-expiration\">\n\t\t\t\t<a class=\"columntitle\"><span>Expiration date<\/span><\/a>\n\t\t\t<\/th>\n\t\t<\/tr>\n\t<\/thead>\n\t<tbody id=\"fileList\">\n\t<\/tbody>\n\t<tfoot>\n\t<\/tfoot>\n<\/table>\n"
            },
            "shareoverview": {
              "id": "shareoverview",
              "content": "<div id='notification'><\/div>\n\n<div id=\"emptycontent\" class=\"hidden\"><\/div>\n\n<input type=\"hidden\" name=\"dir\" value=\"\" id=\"dir\">\n\n<div class=\"nofilterresults hidden\">\n\t<div class=\"icon-search\"><\/div>\n\t<h2>No entries found in this folder<\/h2>\n<\/div>\n\n<table id=\"filestable\" class=\"list-container \">\n\t<thead>\n\t\t<tr>\n\t\t\t<th id='headerName' class=\"hidden column-name\">\n\t\t\t\t<div id=\"headerName-container\">\n\t\t\t\t\t<a class=\"name sort columntitle\" data-sort=\"name\"><span>Name<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t\t<\/div>\n\t\t\t<\/th>\n\t\t\t<th id=\"headerDate\" class=\"hidden column-mtime\">\n\t\t\t\t<a id=\"modified\" class=\"columntitle\" data-sort=\"mtime\"><span>Share time<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t<\/th>\n\t\t\t<th class=\"hidden column-expiration\">\n\t\t\t\t<a class=\"columntitle\"><span>Expiration date<\/span><\/a>\n\t\t\t<\/th>\n\t\t<\/tr>\n\t<\/thead>\n\t<tbody id=\"fileList\">\n\t<\/tbody>\n\t<tfoot>\n\t<\/tfoot>\n<\/table>\n"
            },
            "systemtagsfilter": {
              "id": "systemtagsfilter",
              "content": "<div id=\"controls\">\n<\/div>\n\n<div id=\"emptycontent\" class=\"hidden\">\n\t<div class=\"icon-folder\"><\/div>\n\t<h2>No files in here<\/h2>\n\t<p class=\"uploadmessage hidden\"><\/p>\n<\/div>\n\n<div class=\"nofilterresults emptycontent hidden\">\n\t<div class=\"icon-search\"><\/div>\n\t<h2>No entries found in this folder<\/h2>\n\t<p><\/p>\n<\/div>\n\n<table id=\"filestable\" data-preview-x=\"250\" data-preview-y=\"250\">\n\t<thead>\n\t\t<tr>\n\t\t\t<th id='headerName' class=\"hidden column-name\">\n\t\t\t\t<div id=\"headerName-container\">\n\t\t\t\t\t<a class=\"name sort columntitle\" data-sort=\"name\"><span>Name<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t\t<\/div>\n\t\t\t<\/th>\n\t\t\t<th id=\"headerSize\" class=\"hidden column-size\">\n\t\t\t\t<a class=\"size sort columntitle\" data-sort=\"size\"><span>Size<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t<\/th>\n\t\t\t<th id=\"headerDate\" class=\"hidden column-mtime\">\n\t\t\t\t<a id=\"modified\" class=\"columntitle\" data-sort=\"mtime\"><span>Modified<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t<\/th>\n\t\t<\/tr>\n\t<\/thead>\n\t<tbody id=\"fileList\">\n\t<\/tbody>\n\t<tfoot>\n\t<\/tfoot>\n<\/table>\n<input type=\"hidden\" name=\"dir\" id=\"dir\" value=\"\" \/>\n\n"
            },
            "trashbin": {
              "id": "trashbin",
              "content": "<div id=\"controls\">\n\t<div id=\"file_action_panel\"><\/div>\n<\/div>\n<div id='notification'><\/div>\n\n<div id=\"emptycontent\" class=\"hidden\">\n\t<div class=\"icon-delete\"><\/div>\n\t<h2>No deleted files<\/h2>\n\t<p>You will be able to recover deleted files from here<\/p>\n<\/div>\n\n<input type=\"hidden\" name=\"dir\" value=\"\" id=\"dir\">\n\n<div class=\"nofilterresults hidden\">\n\t<div class=\"icon-search\"><\/div>\n\t<h2>No entries found in this folder<\/h2>\n\t<p><\/p>\n<\/div>\n\n<table id=\"filestable\" class=\"list-container \">\n\t<thead>\n\t\t<tr>\n\t\t\t<th id=\"headerSelection\" class=\"hidden column-selection\">\n\t\t\t\t<input type=\"checkbox\" id=\"select_all_trash\" class=\"select-all checkbox\"\/>\n\t\t\t\t<label for=\"select_all_trash\">\n\t\t\t\t\t<span class=\"hidden-visually\">Select all<\/span>\n\t\t\t\t<\/label>\n\t\t\t<\/th>\n\t\t\t<th id='headerName' class=\"hidden column-name\">\n\t\t\t\t<div id=\"headerName-container\">\n\t\t\t\t\t<a class=\"name sort columntitle\" data-sort=\"name\"><span>Name<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t\t\t<span id=\"selectedActionsList\" class='selectedActions'>\n\t\t\t\t\t\t<a href=\"\" class=\"actions-selected\">\n\t\t\t\t\t\t\t<span class=\"icon icon-more\"><\/span>\n\t\t\t\t\t\t\t<span>Actions<\/span>\n\t\t\t\t\t\t<\/a>\n\t\t\t\t\t<\/span>\n\t\t\t\t<\/div>\n\t\t\t<\/th>\n\t\t\t<th id=\"headerDate\" class=\"hidden column-mtime\">\n\t\t\t\t<a id=\"modified\" class=\"columntitle\" data-sort=\"mtime\"><span>Deleted<\/span><span class=\"sort-indicator\"><\/span><\/a>\n\t\t\t<\/th>\n\t\t<\/tr>\n\t<\/thead>\n\t<tbody id=\"fileList\">\n\t<\/tbody>\n\t<tfoot>\n\t<\/tfoot>\n<\/table>\n"
            }
          },
          "hiddenFields": [
            
          ]
        }
      ]
    },
    {
      "file": "\/tmp\/nextcloud\/lib\/private\/AppFramework\/Http\/Dispatcher.php",
      "line": 119,
      "function": "render",
      "class": "OCP\\AppFramework\\Http\\TemplateResponse",
      "type": "->",
      "args": [
        
      ]
    },
    {
      "file": "\/tmp\/nextcloud\/lib\/private\/AppFramework\/App.php",
      "line": 118,
      "function": "dispatch",
      "class": "OC\\AppFramework\\Http\\Dispatcher",
      "type": "->",
      "args": [
        {
          "__class__": "OCA\\Files\\Controller\\ViewController"
        },
        "index"
      ]
    },
    {
      "file": "\/tmp\/nextcloud\/lib\/private\/AppFramework\/Routing\/RouteActionHandler.php",
      "line": 47,
      "function": "main",
      "class": "OC\\AppFramework\\App",
      "type": "::",
      "args": [
        "ViewController",
        "index",
        {
          "__class__": "OC\\AppFramework\\DependencyInjection\\DIContainer"
        },
        {
          "_route": "files.view.index"
        }
      ]
    },
    {
      "function": "__invoke",
      "class": "OC\\AppFramework\\Routing\\RouteActionHandler",
      "type": "->",
      "args": [
        {
          "_route": "files.view.index"
        }
      ]
    },
    {
      "file": "\/tmp\/nextcloud\/lib\/private\/Route\/Router.php",
      "line": 297,
      "function": "call_user_func",
      "args": [
        {
          "__class__": "OC\\AppFramework\\Routing\\RouteActionHandler"
        },
        {
          "_route": "files.view.index"
        }
      ]
    },
    {
      "file": "\/tmp\/nextcloud\/lib\/base.php",
      "line": 987,
      "function": "match",
      "class": "OC\\Route\\Router",
      "type": "->",
      "args": [
        "\/apps\/files\/"
      ]
    },
    {
      "file": "\/tmp\/nextcloud\/index.php",
      "line": 42,
      "function": "handleRequest",
      "class": "OC",
      "type": "::",
      "args": [
        
      ]
    }
  ],
  "File": "\/tmp\/nextcloud\/lib\/private\/Security\/Normalizer\/IpAddress.php",
  "Line": 79,
  "CustomMessage": "--"
}


```

### Server configuration

**Nextcloud version:**
happened on stable15 and current master

**Updated from an older Nextcloud/ownCloud or fresh install:**
fresh

### PR DESCRIPTION
**Pull request by tcitworld**
Fix #13943

When run with php's build-in server (for instance on `localhost:8080`), IP provided through `$this->server['REMOTE_ADDR']` is `[::1]`, which is not an acceptable format for \inet_pton.

I didn't dig why I've got an IPv6 localhost or anything, but I guess it's still valid.

This removes the brackets if there's any.

### CODE REVIEW

**Comment by kesselb** [comment]
Close #13943

File: lib/private/Security/Normalizer/IpAddress.php [hunk 1]
```diff
@@ -68,6 +68,9 @@ private function getIPv4Subnet(string $ip, int $maskBits = 32): string {
 	 * @return string
 	 */
 	private function getIPv6Subnet(string $ip, int $maskBits = 48): string {
+		if ($ip[0] === '[' && $ip[-1] === ']') { // If IP is with brackets, for example [::1]
```
Comment (kesselb) [inline lib/private/Security/Normalizer/IpAddress.php]: `trim($ip, '[]'))` would also work.

**Review by MorrisJobke — APPROVED** [review]
Code makes sense 👍

**Comment by MorrisJobke** [comment]
@ChristophWurst Mind to review this one?

**Review by ChristophWurst — APPROVED** [review]
Makes sense! Thank you!

**Pull request merged**